ivadomed/scripts/download_data.py

```diff
@@ -208,58 +208,50 @@ def install_data(url, dest_folder, keep=False):
     unzip(tmp_file, extraction_folder)
 
     # Identify whether we have a proper archive or a tarbomb
-    with os.scandir(extraction_folder) as it:
-        has_dir = False
-        nb_entries = 0
-        for entry in it:
-            if entry.name in ("__MACOSX",):
-                continue
-            nb_entries += 1
-            if entry.is_dir():
-                has_dir = True
+    has_dir = False
+    nb_entries = 0
+    for path_object in Path(extraction_folder).iterdir():
+        if path_object.name in ("__MACOSX",):
+            continue
+        nb_entries += 1
+        if path_object.is_dir():
+            has_dir = True
 
     if nb_entries == 1 and has_dir:
         # tarball with single-directory -> go under
-        with os.scandir(extraction_folder) as it:
-            for entry in it:
-                if entry.name in ("__MACOSX",):
-                    continue
-                bundle_folder = entry.path
+        for path_object in Path(extraction_folder).iterdir():
+            if path_object.name in ("__MACOSX",):
+                continue
+            bundle_folder = path_object
     else:
         # bomb scenario -> stay here
-        bundle_folder = extraction_folder
+        bundle_folder = Path(extraction_folder)
 
     # Copy over
-    for cwd, ds, fs in os.walk(bundle_folder):
-        ds.sort()
-        fs.sort()
-        ds[:] = [d for d in ds if d not in ("__MACOSX",)]
-        for d in ds:
-            srcpath = os.path.join(cwd, d)
-            relpath = os.path.relpath(srcpath, bundle_folder)
-            dstpath = os.path.join(dest_folder, relpath)
-            if os.path.exists(dstpath):
-                # lazy -- we assume existing is a directory, otherwise it will crash safely
-                logger.debug("- d- %s", relpath)
-            else:
-                logger.debug("- d+ %s", relpath)
-                os.makedirs(dstpath)
-
-        for f in fs:
-            srcpath = os.path.join(cwd, f)
-            relpath = os.path.relpath(srcpath, bundle_folder)
-            dstpath = os.path.join(dest_folder, relpath)
-            if os.path.exists(dstpath):
+    for path_object in bundle_folder.glob("**/*"):
+        if path_object.is_dir():
+            if path_object.name not in ("__MACOSX",):
+                relpath = path_object.relative_to(bundle_folder)
+                dstpath = Path(dest_folder, relpath)
```

Comment:
VERY NICELY done here. 👍 
I like how versatile the whole pathlib is. 
![image](https://user-images.githubusercontent.com/6720068/128963734-997c12e7-3555-4fe1-9981-4591c2c53bc3.png)